src/libs/DateUtils.ts

```diff
@@ -374,10 +491,172 @@ function getStatusUntilDate(inputDate: string): string {
+        // it's in "hh:mm a" format
+        const tempTime = parse(updatedTime, 'hh:mm a', new Date());
+        if (isValid(tempTime)) {
+            parsedTime = tempTime;
+        }
+    }
+
+    if (!parsedTime) {
+        return '';
+    }
+
+    let parsedDateTime: Date | null = null;
+    if (inputDateTime.includes(':')) {
+        // Check if it includes time
+        const tempDateTime = parse(inputDateTime, 'yyyy-MM-dd HH:mm:ss', new Date());
+        if (isValid(tempDateTime)) {
+            parsedDateTime = tempDateTime;
+        }
+    } else {
+        const tempDateTime = parse(inputDateTime, 'yyyy-MM-dd', new Date());
+        if (isValid(tempDateTime)) {
+            parsedDateTime = tempDateTime;
+        }
+    }
+
+    if (!parsedDateTime) {
+        return '';
+    }
+
+    const updatedDateTime = set(parsedDateTime, {
+        hours: parsedTime.getHours(),
+        minutes: parsedTime.getMinutes(),
+        seconds: parsedTime.getSeconds(),
+    });
+
+    return format(updatedDateTime, 'yyyy-MM-dd HH:mm:ss');
+};
+
+/**
+ * param {String} dateTime in 'HH:mm:ss' format
+ * returns {Object}
+ * example {hour: '11', minute: '10', period: 'AM'}
+ */
+function get12HourTimeObjectFromDate(dateTime: string): {hour: string; minute: string; period: string} {
+    if (!dateTime) {
+        return {
+            hour: '12',
+            minute: '00',
+            period: 'PM',
+        };
+    }
+    const parsedTime = parse(dateTime, 'hh:mm a', new Date());
+    return {
+        hour: format(parsedTime, 'hh'),
+        minute: format(parsedTime, 'mm'),
+        period: format(parsedTime, 'a').toUpperCase(),
+    };
+}
+
+/**
+ * param {String} timeString
+ * returns {String}
+ * example getTimePeriod('11:10 PM') // 'PM'
+ */
+function getTimePeriod(timeString: string): TimePeriod {
+    const parts = timeString.split(' ');
+    return parts[1] as TimePeriod;
+}
+
+/**
+ * param {String} dateTimeStringFirst // YYYY-MM-DD HH:mm:ss
+ * param {String} dateTimeStringSecond // YYYY-MM-DD HH:mm:ss
+ * returns {Boolean}
+ */
+function areDatesIdentical(dateTimeStringFirst: string, dateTimeStringSecond: string): boolean {
+    const date1 = parse(dateTimeStringFirst, 'yyyy-MM-dd HH:mm:ss', new Date());
+    const date2 = parse(dateTimeStringSecond, 'yyyy-MM-dd HH:mm:ss', new Date());
+
+    return isSameSecond(date1, date2);
+}
+
+/**
+ * Checks if the time input is at least one minute in the future.
+ * param {String} timeString: '04:24 AM'
+ * param {String} dateTimeString: '2023-11-14 14:24:00'
+ * returns {Boolean}
+ */
+const isTimeAtLeastOneMinuteInFuture = ({timeString, dateTimeString}: {timeString?: string; dateTimeString: string}): boolean => {
+    let dateToCheck = dateTimeString;
+    if (timeString) {
+        //  return false;
+        // Parse the hour and minute from the time input
+        const [hourStr] = timeString.split(/[:\s]+/);
+        const hour = parseInt(hourStr, 10);
+
+        if (hour === 0) {
```

Comment:
This caused [this bug](https://github.com/Expensify/App/issues/40495): we can't update the custom time to 0h